polkadot/node/core/dispute-coordinator/src/scraping/tests.rs

```diff
@@ -662,3 +662,251 @@ fn inclusions_per_candidate_properly_adds_and_prunes() {
 		assert!(scraper.get_blocks_including_candidate(&candidate.hash()).len() == 0);
 	});
 }
+
+// ----- Inclusions tests -----
+
+#[test]
+fn inclusions_initialization() {
+	let inclusions = Inclusions::new();
+
+	assert!(inclusions.inclusions_inner.is_empty(), "Expected inclusions_inner to be empty");
+	assert!(
+		inclusions.candidates_by_block_number.is_empty(),
+		"Expected candidates_by_block_number to be empty"
+	);
+}
+#[test]
+fn inclusions_insertion() {
+	let mut inclusions = Inclusions::new();
+	let candidate_receipt = make_candidate_receipt(get_magic_candidate_hash());
+	let candidate_hash = candidate_receipt.hash();
+	let block_number = 0;
+	let block_hash = get_block_number_hash(block_number);
+
+	inclusions.insert(candidate_hash, block_number, block_hash);
+
+	// Check inclusions_inner
+	assert!(
+		inclusions.inclusions_inner.contains_key(&candidate_hash),
+		"Expected candidate_hash to be present in inclusions_inner"
+	);
+	let inner_map = inclusions.inclusions_inner.get(&candidate_hash).unwrap();
+	assert!(
+		inner_map.contains_key(&block_number),
```

Comment:
nit: It's worth checking that the len of the Vec is 1 too.

Reply:
Good point. Added len asserts for all the maps including all the inner ones.